### ISSUE TITLE
Add a hook to set @summary on table in XHTML

### ISSUE THREAD

**Issue by robander** [dita-ot/dita-ot#1300]
DITA does not (yet) provide a standard way to provide a table summary, which is often used to enhance accessibility (providing a long text description of the table which is not visible in browsers but can be read by a screen reader). This is close in meaning to the existing <desc> element on tables, except that this content has been used by many to include descriptive text for all users (not just screen readers). Moving it to the summary would hide that text from the browser, and duplicating it actually reduces usability because screen readers will read it twice.

Some users have a requirement to provide this attribute, and currently must do so following some local convention - for example (only an example), using something like this:

``` xml
<desc>Existing table
description<p outputclass="summary">Extended table summary here</p></desc>
```

To enable local conventions, we should add a hook for table and simpletable processing that extensions can catch in order to add summaries. Note that this could also be useful for specialized tables that always have a standard description - for example, the hook could be used to add a standard summary to all <choicetable> elements. I'll update the code so that tables and choicetables include the following line, which (by default) will leave the current empty summary attribute:

``` xml
  <xsl:apply-templates select="." mode="generate-table-summary-attribute"/>
```
